ex03.ex

```diff
@@ -52,20 +13,15 @@ defmodule Ex03 do
 
   (The library functions `Integer.is_even` and `Integer.is_odd` may
   be used if needed.)
-
   """
 
-  def odd_even . . . "your code"
-
-
-  ##############################################################################
-  # 3.2:  5 points #
-  ##################
+  def odd_even([x | t]) when Integer.is_even(x), do: [:even | odd_even(t)]
+  def odd_even([x | t]) when Integer.is_odd(x), do: [:odd | odd_even(t)]
+  def odd_even(x), do: x
```

Comment:
* -2 *

I know you were trying to be thorough by adding this third clause, but by doing so you've actually broken the code.

If I called this function with `[ 1, :cat ]`, the result would be `[:odd, :cat ]`. I'd not get an error. But the description is explicit that the function should take a list of integers. 

One of the golden rules we talked about in the first week is that code should fail if given bad inputs, and not handle it.

Reply:
Good point - thanks